### PR TITLE
Added @node and @revision

### LINKED ISSUE #9013: Add @node and @revision in the questions controller
This issue is a part of #8942 
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [app/controllers/questions_controller.rb](https://github.com/publiclab/plots2/blob/main/app/controllers/questions_controller.rb) in the `plots2` repository (press the little pen Icon) and edit the line as shown below. 

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
def new
+  @node = Node.new
+  @revision = Revision.new
   # use another node body as a template
   node_id = params[:n].to_i
   if node_id && !params[:body] && Node.exists?(node_id)
     node = Node.find(node_id)
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by SavneetKaur01**
Fixes #9013   (<=== Add issue number here)
Thanks!

### CODE REVIEW

File: app/controllers/questions_controller.rb [hunk 1]
```diff
@@ -48,6 +48,8 @@ def index_shadow
 
   # a form for new questions, at /questions/new
   def new
+    @node=Node.new
```
[the remark this answers is not in the document]
Comment (SavneetKaur01) [reply app/controllers/questions_controller.rb:51]: Ok, I will just work on it

**Comment by Manasa2850** [comment]
@SavneetKaur01 could you please mention the issue number here `Fixes #0000 (<=== Add issue number here)`

**Comment by Manasa2850** [comment]
@SavneetKaur01 the corresponding issue is #9013 so could you mention that in `Fixes #0000 (<=== Add issue number here)`
Thanks :)

**Comment by Manasa2850** [comment]
Great work @SavneetKaur01 :tada: 
Welcome to Public Lab :balloon:

**Comment by Manasa2850** [comment]
@cesswairimu could you please review this? I think merging this would solve the issue we're facing in #8943 
Thank you!

**Review by Sagarpreet — APPROVED** [review]
🎉

**Comment by cesswairimu** [comment]
Looks like this breaks the page :thinking: ..screenshot from gitpod 
Screenshot from 2021-01-17 12-38-06

**Comment by Manasa2850** [comment]
Oh I'm really sorry about that @cesswairimu. Not very sure why that's happening. I'll look into it now.
Thank you!

**Comment by cesswairimu** [comment]
no worries @Manasa2850, thought that would fix it too :)

**Comment by Manasa2850** [comment]
@cesswairimu I found out why that page was breaking. It was basically checking for a route when `id=nil` too. So I've fixed it by adding a condition which checks if `id is not nil` and then displays the page only in that case!

```diff
+ <% if params[:id] %>
      <%= form_for :node, :as => :drupal_node, :url => url do |f| %>
       <%= render partial: 'layouts/errorMessages', locals: { model: @node } %>
      <% end %>

       <%= form_for :revision, :as => :revision, :url => url, :html => {:class => "form well legacy-form row"} do |f| %>
        <%= render partial: 'layouts/errorMessages', locals: { model: @revision } %>
       <% end %>
+ <% end %>
```

**Comment by jywarren** [comment]
Great solution @Manasa2850 - i love it. I'll make it as a suggestion on this PR, but will also invite you to be a collaborator so you can do reviews too. Much appreciated!

**Comment by SavneetKaur01** [comment]
@Manasa2850 I will add it to this PR if I may.

**Comment by Manasa2850** [comment]
@jywarren sure. Thanks a ton! I would love to be a collaborator :tada:

**Comment by Manasa2850** [comment]
@SavneetKaur01 please go ahead. Thank you!

**Review by jywarren — APPROVED** [review]
Looks great! Can you try this out and upload a screenshot on https://stable.publiclab.org once it finishes building? Thank you!!!

**Comment by cesswairimu** [comment]
:tada: :tada:

**Pull request merged**